core/server/data/meta/blog_logo.js

```diff
@@ -1,33 +1,43 @@
 var utils            = require('../../utils'),
-    settingsCache    = require('../../settings/cache');
+    settingsCache    = require('../../settings/cache'),
+    blogIconUtils    = require('../../utils/blog-icon'),
+    config           = require('../../config'),
+    path             = require('path');
 
 function getBlogLogo() {
-    var logo = {};
+    var logo = {},
+        filePath;
 
     if (settingsCache.get('logo')) {
         logo.url = utils.url.urlFor('image', {image: settingsCache.get('logo')}, true);
     } else {
         // CASE: no publication logo is updated. We can try to use either an uploaded publication icon
         // or use the default one to make
         // Google happy with it. See https://github.com/TryGhost/Ghost/issues/7558
-        if (settingsCache.get('icon')) {
-            // CASE: we have a custom publication icon
-            logo.url = utils.url.urlFor('image', {image: settingsCache.get('icon')}, true);
-            // TODO: make a util that checks for `ico` file extension
-            if (logo.url.match(/\.ico$/i)) {
-                // CASE: we have an `.ico` file extension. `image-size` util can't fetch dimensions
-                // for this file extension
-                logo.dimensions = {
-                    width: 60,
-                    height: 60
-                };
+        logo.url = blogIconUtils.getIconUrl(true);
+        if (blogIconUtils.isIcoImageType(logo.url)) {
+            if (settingsCache.get('icon')) {
+                filePath = settingsCache.get('icon').replace(new RegExp('^' + utils.url.getSubdir() + '/' + utils.url.STATIC_IMAGE_URL_PREFIX), '');
+                filePath = path.join(config.getContentPath('images'), filePath);
+            } else {
+                filePath = path.join(config.get('paths:publicFilePath'), 'favicon.ico');
             }
-        } else {
-            logo.url = utils.url.urlFor({relativeUrl: '/favicon.ico'}, true);
+            // For now, return hard coded values
             logo.dimensions = {
                 width: 60,
                 height: 60
             };
+            // TODO: make this work with getIconDimensions to fetch the real ico sizes
```

Comment:
I didn't manage to get this work properly and was asking myself in the end if this is maybe to much effort for a fallback scenario to make Google happy. 

Because `getIconDimensions` is async, I would need to wrap the whole fn here in a Promise as well, which has an impact on `/meta/index.js`...

But maybe there's a completely different way, which I didn't see yet?

Any help / advice is highly appreciated here @ErisDS, @kirrg001